### PR TITLE
Span Events Java Documentation

### PR DESCRIPTION
**Pull request by link04**
### What does this PR do? What is the motivation?

Adds the documentation for Java Span Events.
Follow up to https://github.com/DataDog/documentation/pull/25100, where it couldn't be documented until the feature was completed for Java.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/tracing/trace_collection/custom_instrumentation/java/otel.md [hunk 1]
```diff
@@ -181,6 +181,55 @@ public class Example {
 }
 ```
 
+## Adding span events
+
+_Minimum SDK version: 1.40.0._
```
Comment (urseberry) [inline content/en/tracing/trace_collection/custom_instrumentation/java/otel.md:186]:
```suggestion
<div class="alert alert-info">Adding span events requires SDK version 1.40.0 or higher</div>
```

File: content/en/tracing/trace_collection/custom_instrumentation/java/otel.md [hunk 2]
```diff
@@ -181,6 +181,55 @@ public class Example {
...
+You can add span events using the `addEvent` API. This method requires a `name` parameter and optionally accepts `attributes` and `timestamp` parameters. The method creates a new span event with the specified properties and associates it with the corresponding span.
+
+- **Name** [_required_]: A string representing the event's name.
+- **Attributes** [_optional_]: Zero or more key-value pairs with the following properties:
+  - The key must be a non-empty string.
+  - The value can be either:
+    - A primitive type: string, Boolean, or number.
+    - A homogeneous array of primitive type values (for example, an array of strings).
+  - Nested arrays and arrays containing elements of different data types are not allowed.
+- **Timestamp** [_optional_]: A UNIX timestamp representing the event's occurrence time, expects an `Instant` object.
```
Comment (urseberry) [inline content/en/tracing/trace_collection/custom_instrumentation/java/otel.md:197]:
```suggestion
- **Timestamp** [_optional_]: A UNIX timestamp representing the event's occurrence time. Expects an `Instant` object.
```

File: content/en/tracing/trace_collection/custom_instrumentation/java/otel.md [hunk 3]
```diff
@@ -181,6 +181,55 @@ public class Example {
...
+          .put(AttributeKey.stringArrayKey("string_array"), Arrays.asList("5", "6"))
+          .put(AttributeKey.booleanArrayKey("bool_array"), Arrays.asList(true, false))
+          .build();
+
+      span.addEvent("Event With No Attributes");
+      span.addEvent("Event With Some Attributes", eventAttributes);
+
+      span.recordException(new Exception("Error Message"));
+      span.recordException(new Exception("Error Message"), 
+          Attributes.builder().put(AttributeKey.stringKey("status"), "failed").build());
```
Comment (urseberry) [inline content/en/tracing/trace_collection/custom_instrumentation/java/otel.md:215]:
```suggestion
Attributes eventAttributes = Attributes.builder()
     .put(AttributeKey.longKey("int_val"), 1L)
     .put(AttributeKey.stringKey("string_val"), "two")
     .put(AttributeKey.longArrayKey("int_array"), Arrays.asList(3L, 4L))
     .put(AttributeKey.stringArrayKey("string_array"), Arrays.asList("5", "6"))
     .put(AttributeKey.booleanArrayKey("bool_array"), Arrays.asList(true, false))
     .build();

span.addEvent("Event With No Attributes");
span.addEvent("Event With Some Attributes", eventAttributes);

span.recordException(new Exception("Error Message"));
span.recordException(new Exception("Error Message"), 
    Attributes.builder().put(AttributeKey.stringKey("status"), "failed").build());
```

**Review by urseberry — COMMENTED** [review]
Thank you so much for adding these instructions!

File: content/en/tracing/trace_collection/custom_instrumentation/java/otel.md (hunk 3 above)
Comment (link04) [reply content/en/tracing/trace_collection/custom_instrumentation/java/otel.md]: Than you! Updated in https://github.com/DataDog/documentation/pull/25157/commits/9e2c256728da10da5a7a200c66fbcd8463d9c43a.

File: content/en/tracing/trace_collection/custom_instrumentation/java/otel.md (hunk 2 above)
Comment (link04) [reply content/en/tracing/trace_collection/custom_instrumentation/java/otel.md]: Than you! Updated in https://github.com/DataDog/documentation/pull/25157/commits/9e2c256728da10da5a7a200c66fbcd8463d9c43a.

File: content/en/tracing/trace_collection/custom_instrumentation/java/otel.md (hunk 1 above)
Comment (link04) [reply content/en/tracing/trace_collection/custom_instrumentation/java/otel.md]: Than you! Added and updated related docs in https://github.com/DataDog/documentation/pull/25157/commits/9e2c256728da10da5a7a200c66fbcd8463d9c43a.

**Review by aliciascott — APPROVED** [review]
thanks for the updates looks good!

**Pull request merged**